app/src/main/java/ro/code4/deurgenta/ui/backpack/main/BackpacksFragment.kt

```diff
@@ -0,0 +1,90 @@
+import kotlinx.android.synthetic.main.fragment_backpacks.*
+import org.koin.android.viewmodel.ext.android.sharedViewModel
+import org.parceler.Parcels
+import ro.code4.deurgenta.R
+import ro.code4.deurgenta.data.model.Backpack
+import ro.code4.deurgenta.ui.base.ViewModelFragment
+
+class BackpacksFragment : ViewModelFragment<BackpacksViewModel>() {
+    override val screenName: Int
+        get() = R.string.analytics_title_backpack
+    override val layout: Int
+        get() = R.layout.fragment_backpacks
+    override val viewModel: BackpacksViewModel by sharedViewModel(from = { requireParentFragment() })
+    private val backpacksAdapter: BackpacksAdapter by lazy {
+        BackpacksAdapter(requireContext()) {
+            if (it !== Backpack.EMPTY) {
+                findNavController().navigate(
+                    R.id.action_backpackMainFragment_to_backpackDetailsFragment,
+                    bundleOf(BackpackDetailsFragment.KEY_BACKPACK to Parcels.wrap(it))
+                )
+            }
+        }
+    }
+    private var loadingAnimator: ObjectAnimator? = null
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModel.fetchBackpacks()
+
+        add_new_backpack.setOnClickListener {
+            findNavController().navigate(R.id.action_backpackMainFragment_to_newBackpackDialogFragment)
+        }
+
+        with(remainder) {
+            val underlineAction = SpannableString(getString(R.string.backpack_btn_reminder))
+            underlineAction.setSpan(UnderlineSpan(), 0, underlineAction.length, Spanned.SPAN_INCLUSIVE_INCLUSIVE)
+            text = underlineAction
+            setOnClickListener { findNavController().navigate(R.id.action_backpackMainFragment_to_homeFragment) }
+        }
+        with(backpacks_list) {
+            layoutManager = LinearLayoutManager(requireContext())
+            addItemDecoration(DividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL))
+            adapter = backpacksAdapter
+        }
+
+        viewModel.uiModel.observe(viewLifecycleOwner) {
+            when (it) {
+                is Error -> TODO()
+                Loading -> setAsLoading(true)
+                is Success -> {
+                    setAsLoading(false)
+                    backpacksAdapter.submitList(it.backpacks)
+                }
+            }
+        }
+    }
+
+    private fun setAsLoading(isLoading: Boolean) {
+        loadingIndicator.visibility = if (isLoading) View.VISIBLE else View.GONE
+        if (isLoading) {
+            loadingAnimator?.cancel()
+            loadingAnimator = ObjectAnimator.ofFloat(loadingIndicator, "rotation", 0f, 360f).apply {
+                repeatCount = ObjectAnimator.INFINITE
+                duration = 2000
+                interpolator = LinearInterpolator()
+                repeatMode = ObjectAnimator.RESTART
+            }
```

Comment:
maybe we could extract a bit of the loading animation and reuse it?

Reply:
I extracted the ObjectAnimator creation in an extension method to be used in the splash and backpack related screens.